### DISCUSSION TITLE
Message "HAXM is not installed" when i try to run DOS applications

### DISCUSSION THREAD

**Discussion by ZeroGit2015** [leecher1337/ntvdmx64#217] [General]
I downloaded installation of ntvdm64 from url http://www.columbia.edu/~em36/ntvdmx64-haxm-fre.7z , extract files and run ru/install.bat with administration rights. There was no errors. But when i try to run any DOS application i see message "HAXM is not installed". :( How i can fix it?

**Comment by leecher1337** [comment]
What do you get when you run

```
net start IntelHaxmNtvdm
```

on elevated command prompt? If driver is installed but fails to start, check event log for possible errors (prereq is compatible CPU and deactivated Hyper-V as specified in haxm.txt documentation).

**Reply by ZeroGit2015** [reply]
> What do you get when you run
> 
> ```
> net start IntelHaxmNtvdm
> ```
My OS Windows 10 22H2 x64: 
I see:
System error 2148204812. The certificate has been revoked by the provider of this certificate

**Reply by leecher1337** [reply]
Bad luck then, seems that the suckers of M$ revoked the leaked NVIDIA code signing certificate and so it no longer works with Windows 11.  Maybe you have luck and this registry key works: https://gist.github.com/wdormann/4ab5f808bc6ac22f4aadd2b037690717
Would be interesting if this works. As far as I understand, blocked driver certificate hashes are stored in C:\Windows\System32\CodeIntegrity\driver.stl but I guess there is no way to revert this file back to state of 2019.

If registry key does not work, you may need to use Windows in Test mode, [(refer to the instructions in haxm.txt](https://github.com/leecher1337/ntvdmx64/blob/11a1463b65a60d59a542c9c793af1616d444ba59/ntvdmpatch/doc/haxm.txt#L103).

**Reply by ZeroGit2015** [reply]
- Import https://gist.github.com/wdormann/4ab5f808bc6ac22f4aadd2b037690717 in registry had no results

- bcdedit /set hypervisorlaunchtype off + bcdedit /set testsigning on had no results

Message didn`t change: System error 2148204812. The certificate has been revoked by the provider of this certificate

May be i must reinstall ntvdm64 or install last version of Haxm now?

**Reply by leecher1337** [reply]
If it also complains in Test mode about certificate, you can try to install the old .sys driver instead that wasn't signed with the now blacklisted certificate: https://github.com/leecher1337/ntvdmx64/blob/1986e1852f2f7a04a2c7f668ac8aa1517f092409/ntvdmpatch/haxm/IntelHaxm.sys
Driver is the same, just other test-signing (not productive) signature that shouldn't be blacklisted.

It may be enough to 

1) download https://github.com/leecher1337/ntvdmx64/blob/1986e1852f2f7a04a2c7f668ac8aa1517f092409/ntvdmpatch/haxm/IntelHaxm.sys
2) replace %windir%\system32\drivers\IntelHaxmNtvdm.sys with the file mentioned above (obviously renamed to IntelHaxmNtvdm.sys ). 
3) Download https://github.com/leecher1337/ntvdmx64/blob/32688f914966bd8c2a83c2bc9df7ee9abaac9687/ntvdmpatch/haxm/IntelHaxm.cer
4) `certutil -addstore "TrustedPublisher" IntelHaxm.cer`

And then try to start the driver again with net start again.

**Reply by ZeroGit2015** [reply]
It works! Very tnx! Does it work if windows is on the test mode only?

**Reply by leecher1337** [reply]
Yes, unfortunately only works in test mode, as the productive certificate was revoked, so we have to resort to test mode. M$ made it nearly impossible to obtain a valid code signing certificate, they charge you insane amounts of money for it (for just 1 year), plus you have to be a registered company to even get one and you have to submit every driver to them for signing (as they now also disabled cross-signing), so they made it practically impossible for opensource projects to build drivers.
I also asked intel if they could pull my enhancements of the driver into their main signed driver, but they refused to do so, unfortunately, even though there were no breaking changes, just enhancements.
So my last resort was using a leaked driver certificate, but it was blacklisted now, as it seems. 
So Test mode is the only possibility unless we find some company that would sign the driver,.

**Reply by ZeroGit2015** [reply]
Oh. I understood. Thank you for help.
